tools-local/unix/tool-helper.sh

```diff
@@ -0,0 +1,205 @@
+        exit 1
+    fi
+
+    if [ -z "$2" ]; then
+        echo "Argument passed as tool name is empty. Please provide a non-empty string."
+        exit 1
+    fi
+
+    repoRoot="$1"
+    toolName="$2"
+    . "$repoRoot/.toolversions"
+
+    # Evaluate toolName. This assigns the metadata of toolName to tools.
+    eval "tools=\$$toolName"
+
+    # Evaluate tools. Each argument here is tool specific data such as DeclaredVersion of toolName.
+    eval "$tools"
+}
+
+# Gets the value corresponding to the specified configuration from the .toolversions file.
+# Exit 1 if the value is not found or empty.
+get_tool_config_value()
+{
+    if [ -z "$1" ]; then
+        echo "Argument passed as repository-root is empty. Please provide a non-empty string."
+        exit 1
+    fi
+
+    if [ -z "$2" ]; then
+        echo "Argument passed as tool name is empty. Please provide a non-empty string."
+        exit 1
+    fi
+
+    if [ -z "$3" ]; then
+        echo "Argument passed as configuration name is empty. Please provide a non-empty string."
+        exit 1
+    fi
+
+    repoRoot="$1"
+    toolName="$2"
+    configName="$3"
+    configValue="$(eval_tool "$repoRoot" "$toolName"; eval echo "\$$configName")"
+
+    if [ -z "$configValue" ]; then
+        echo "Unable to read the value corresponding to $configName from the .toolversions file."
+        exit 1
+    fi
+
+    echo "$configValue"
+}
+
+# Gets the name of the download package corresponding to the specified tool name.
+# Download package name is read from the .toolversions file.
+# Exit 1 if unable to read the name of the download package from the .toolversions file.
+get_download_package_name()
+{
+    if [ -z "$1" ]; then
+        echo "Argument passed as repository-root is empty. Please provide a non-empty string."
+        exit 1
+    fi
+
+    if [ -z "$2" ]; then
+        echo "Argument passed as tool name is empty. Please provide a non-empty string."
+        exit 1
+    fi
+
+    repoRoot="$1"
+    toolName="$2"
+    osName="$(get_os_name)"
+    get_tool_config_value "$repoRoot" "$toolName" "DownloadFile$osName"
+}
+
+# Gets the absolute path to the local cache corresponding to the specified tool.
+# Path is read from the .toolversions file.
+get_local_tool_folder()
+{
+    if [ -z "$1" ]; then
+        echo "Argument passed as repository-root is empty. Please provide a non-empty string."
+        exit 1
+    fi
+
+    if [ -z "$2" ]; then
+        echo "Argument passed as tool name is empty. Please provide a non-empty string."
+        exit 1
+    fi
+
+    repoRoot="$1"
+    toolName="$2"
+    toolFolder="$(get_tool_config_value "$repoRoot" "$toolName" "LocalToolFolder")"
+
+    case "$toolFolder" in
+        /*)
+            echo "$toolFolder"
+            ;;
+        *)
+            # Assumed that the path specified in .toolversion is relative to the repository root.
+            echo "$repoRoot/$toolFolder"
+        ;;
```

Comment:
Indented.

---
In reply to: [111210675](https://github.com/ravimeda/corefx/pull/9#discussion_r111210675) [](ancestors = 111210675)